### PR TITLE
#158128651 User sign up endpoint with jwt and sign in endpoint with tests

### PR DESCRIPTION
**Pull request by RuthNjeri**
**What does this PR do?**
1. Create user using an endpoint.
2. Sign in user with an endpoint including jwt
3. Test the sign up and sign in endpoint including edge cases.
**Description of task completed?**
1. Sign up endpoint `/auth/signup`
2. Sign in endpoint `/auth/login`
4. jwt authentication and decoding.
5. Test jwt authentication
3. Test to check if a user can be created with the sign up endpoint
4. Test to check if a user can sign in and a token is created
5. Test if the user can sign in with the right credentials
6. Test if a user in already created can be signed up again.

**How to test the project**
Clone the project
`git clone https://github.com/RuthNjeri/Maintenance-Tracker-Application.git`

Create a virtual environment.
`virtualenv project-env`
Activate the virtual environment
`source project-env/bin/activate`
`pip install requirements.txt`
run the test file
`pytest`

What is the relevant pivotal tracker stories?

[#158128651](https://www.pivotaltracker.com/story/show/158128651)

### CODE REVIEW

**Comment by ghost** [comment]
Good login logic.

**Comment by johnmusiu** [comment]
Comprehensive testing for auth.

File: project/users/views.py [hunk 1]
```diff
@@ -1,57 +1,101 @@
...
-    for u in users_list:
-        if u['email'] == email and u['password'] == password:
-            global logged_in
-            logged_in = u['email']
-            return jsonify({'logged_in': True}),200
-    return jsonify({'logged_in': False}),400 
+@users.route('/auth/signup', methods=['POST'])
+def create_user():
+    form = request.get_json()
+    print('form', form)
```
Comment (valeria-chemtai) [inline project/users/views.py:56]: There is no need for you to have print statements in your work if they're not adding any value to it.
